### PR TITLE
Config reloader pod ordinal number

### PR DESCRIPTION
**Pull request by vsliouniaev**
This PR introduces a new `STATEFULSET_ORDINAL_NUMBER` environment variable in the prometheus config reloader, by regexing the numbers from the end of  `POD_NAME`. This allows a prometheus configuration with
```
externalLabels:
  prometheus_replica: $(STATEFULSET_ORDINAL_NUMBER)
```

The motivation for this is our current setup:
- Multiple prometheuses running in the same cluster, deployed into the same namespace
- All write to the same long-term storage database
- Remote-read from the database is used to execute rules, pulling data from multiple prometheuses, but requiring distinction between the replicas

### CODE REVIEW

**Comment by brancz** [comment]
Could you elaborate why the current set of external label replacement is not sufficient for this? For differentiation it already suffices for your use case as far as I can tell.

**Comment by vsliouniaev** [comment]
I need to have a consistent external label value, that does not change with the name of the `prometheus` but is still able to differentiate between replicas. The key of the label itself doesn't matter, but the value needs to be consistent between separate operated prometheuses

File: cmd/prometheus-config-reloader/main_test.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package main
```
Comment (brancz) [inline cmd/prometheus-config-reloader/main_test.go]: this needs a license header

**Comment by brancz** [comment]
That makes sense. I'm ok with this.

**Comment by brancz** [comment]
Thanks!

**Comment by caarlos0** [comment]
are there any plans for a new release soon?

**Comment by brancz** [comment]
There should be one happening some time next week.

**Comment by caarlos0** [comment]
awesome, thanks!

**Pull request merged**